api/auth.go

```diff
@@ -0,0 +1,155 @@
+package api
+
+import (
+	"context"
+	"fmt"
+	"github.com/golang-jwt/jwt/v4"
+	"golang.org/x/crypto/bcrypt"
+	"gym_management_system/db"
+	"gym_management_system/errors"
+	"log"
+	"net/http"
+	"os"
+	"time"
+)
+
+func validPassword(reqPw, encPw string) bool {
+	return bcrypt.CompareHashAndPassword([]byte(encPw), []byte(reqPw)) == nil
+}
+
+func HashPassword(password string) (string, error) {
+	pw, err := bcrypt.GenerateFromPassword([]byte(password), bcrypt.DefaultCost)
+	return string(pw), err
+}
+
+func createJWT(account *db.Account, exp time.Time) (string, error) {
+	claims := &Claims{
+		Id: account.Id,
+		RegisteredClaims: jwt.RegisteredClaims{
+			ExpiresAt: jwt.NewNumericDate(exp),
+		},
+	}
+	token := jwt.NewWithClaims(jwt.SigningMethodHS256, claims)
+	secret := os.Getenv("JWT_SECRET")
+	return token.SignedString([]byte(secret))
+}
+
+func validateJWT(tokenString string, claims jwt.Claims) (*jwt.Token, error) {
+	secret := os.Getenv("JWT_SECRET")
+	log.Println(secret)
+	log.Println("validating jwt")
+	return jwt.ParseWithClaims(tokenString, claims, func(token *jwt.Token) (any, error) {
+		if _, ok := token.Method.(*jwt.SigningMethodHMAC); !ok {
+			return nil, fmt.Errorf("unexpected signing method: %v", token.Header["alg"])
+		}
+		return []byte(secret), nil
+	})
+}
+
+func withJWTAuth(next http.Handler) http.Handler {
+	log.Println("jwt auth middleware")
```

Comment:
You can add logging middleware to your router that always logs what endpoints it hits. This way you could drop the initial logging you have in your handlers.